docs/user/filter_expressions.rst

```diff
@@ -0,0 +1,205 @@
+Filter Expressions
+==================
+Filter expressions describe which users a :ref:`recipe <recipes>` should be
+executed for. They're executed locally in the client's browser and, if they
+pass, the corresponding recipe is executed. Filter expressions have access to
+information about the user, such as their location, locale, and Firefox version.
+
+Filter expressions are written using a language called JEXL_. JEXL is an
+open-source expression language that is given a context (in this case,
+information about the user's browser) and evaluates a statement using that
+context. JEXL stands for "JavaScript Expression Language" and uses JavaScript
+syntax for several (but not all) of its features.
+
+.. note:: The rest of this document includes examples of JEXL syntax that has
+   comments inline with the expressions. JEXL does **not** have any support for
+   comments in statements, but we're using them to make understanding our
+   examples easier.
+
+.. _JEXL: https://github.com/TechnologyAdvice/Jexl
+
+JEXL Basics
+-----------
+The `JEXL Readme`_ describes the syntax of the language in detail; the following
+section covers the basics of writing valid JEXL expressions.
+
+.. note:: Normally, JEXL doesn't allow newlines or other whitespace besides
+   spaces in expressions, but filter expressions in Normandy allow arbitrary
+   whitespace.
+
+A JEXL expression evaluates down to a single value. JEXL supports several basic
+types, such as numbers, strings (single or double quoted), and booleans. JEXL
+also supports several operators for combining values, such as arithmetic,
+boolean operators, comparisons, and string concatenation.
+
+.. code-block:: javascript
+
+   // Arithmetic
+   2 + 2 - 3 // == 1
+
+   // Numerical comparisons
+   5 > 7 // == false
+
+   // Boolean operators
+   false || 5 > 4 // == true
+
+   // String concatenation
+   "Mozilla" + " " + "Firefox" // == "Mozilla Firefox"
+
+Expressions can be grouped using parenthesis:
+
+.. code-block:: javascript
+
+   ((2 + 3) * 3) - 3 // == 7
+
+JEXL also supports lists and objects (known as dictionaries in other languages)
+as well as attribute access:
+
+.. code-block:: javascript
+
+   [1, 2, 1].length // == 3
+   {foo: 1, bar: 2}.foo // == 1
+
+Unlike JavaScript, JEXL supports an ``in`` operator for checking if a substring
+is in a string or if an element is in an array:
+
+.. code-block:: javascript
+
+   "bar" in "foobarbaz" // == true
+   3 in [1, 2, 3, 4] // == true
+
+The context passed to JEXL can be expressed using identifiers, which also
+support attribute access:
+
+.. code-block:: javascript
+
+   normandy.locale == 'en-US' // == true if the client's locale is en-US
+
+Another unique feature of JEXL is transforms, which modify the value given to
+them. Transforms are applied to a value using the ``|`` operator, and may take
+additional arguments passed in the expression:
+
+.. code-block:: javascript
+
+   'January 7th, 1980'|date // == a date object
```

Comment:
This won't work, right now.

```
> new Date('January 7th, 1980')
Invalid Date
```

The values that work right now are, according the JS spec, ISO8601 format, or whatever the browser feels like today. Probably best to stick with ISO8601, like `1980-01-07`.